system_metrics_collector/src/system_metrics_collector/cpu_data.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SYSTEM_METRICS_COLLECTOR__CPU_DATA_HPP_
+#define SYSTEM_METRICS_COLLECTOR__CPU_DATA_HPP_
+
+#include <string>
+#include <sstream>
+
+/**
+ * Struct containing data parsed from a /proc/stat cpu line
+ */
+struct ProcCpuData
```

Comment:
```suggestion
class ProcCpuData
```

Let's avoid using struct for non-POD types. This is confusing for readers who usually expect `struct` objects to be record-like structures, not full blown classes like this one with a ctor, etc.